saleor/discount/utils/order.py

```diff
@@ -48,19 +53,56 @@ def create_or_update_discount_objects_from_promotion_for_order(
     lines_info: list["EditableOrderLineInfo"],
     database_connection_name: str = settings.DATABASE_CONNECTION_DEFAULT_NAME,
 ):
-    create_order_line_discount_objects_for_catalogue_promotions(lines_info)
-    # base unit price must reflect all actual catalogue discounts
+    delete_order_line_discount_objects_for_catalogue_promotions(lines_info)
```

Comment:
As we operate on denoramilzed value, what is the reason for calling this each time when `fetch_order_prices_if_expired` is called?
I mean, any logic related to order_promotion or catalogue_promotion is applicable when creating new line for order, right? In that case we don't need to call this when we need to re-calculate prices, right?
Also, I've checked for example `OrderLineDiscountUpdate`. When calling it, we create new `OrderLineDiscount` object. As MANUAL overwrites the catalogue promotion, this should happen there IMHO. Right now, some part of the logic happens there [ `OrderLineDiscountUpdate`] (adding MANUAL), and some part of the logic here (delete CATALOGUE).

Reply:
It is here to make sure we won't endup with two discounts which shouldn't be combined. But maybe you are right, it should be done only when handling manual line discount and gift.

> I mean, any logic related to order_promotion or catalogue_promotion is applicable when creating new line for order, right?
This is not true for order_promotion.

Reply:
> This is not true for order_promotion.

I forgot about this one 😄 

But still, I think that discount objects re-creation should be triggered when we know that something has been changed. By "Something" I mean the action that would have an impact on base_price/total/discount objects. 

We can also handle this as a separate task, but please discuss it with your team. Right now, some logic for creating discount objects is called inside the mutation/adding order line, and some of the logic is called inside `fetch_order_prices_if_expired`.